staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/clientset/versioned/clientset.go

```diff
@@ -51,9 +53,17 @@ func (c *Clientset) Discovery() discovery.DiscoveryInterface {
 }
 
 // NewForConfig creates a new Clientset for the given config.
+// If config's RateLimiter is not set and QPS and Burst are acceptable,
+// NewForConfig will generate a shared rate-limiter for config.
 func NewForConfig(c *rest.Config) (*Clientset, error) {
 	configShallowCopy := *c
-	if configShallowCopy.RateLimiter == nil && configShallowCopy.QPS > 0 {
+	if configShallowCopy.RateLimiter == nil && configShallowCopy.QPS < 0 {
+		return nil, fmt.Errorf("QPS is required to be greater than or equal to 0 when RateLimiter is not set")
+	}
+	if configShallowCopy.RateLimiter == nil && configShallowCopy.Burst < 0 {
+		return nil, fmt.Errorf("Burst is required to be greater than or equal to 0 when RateLimiter is not set")
+	}
+	if configShallowCopy.RateLimiter == nil && configShallowCopy.QPS > 0 && configShallowCopy.Burst > 0 {
```

Comment:
the case I actually wondered about erroring on QPS > 0 and Burst == 0. If that hangs today, then erroring would make sense to me (we're not causing anything that is currently working to fail, and it makes the failure more obvious)

Reply:
got it